lib/main.js

```diff
@@ -741,6 +741,19 @@ function GetState(state) {
     MML.SUPER.ID = ID = 0;
     MathJax.OutputJax.CommonHTML.ID = 0;
   }
+
+  // Clear any existing user defined macros, then load macros from state
+  Object.keys(TEX.Definitions.macros).forEach(function(macroName){
+    if (TEX.Definitions.macros[macroName].isUser) {
+      delete TEX.Definitions.macros[macroName];
+    }
+  });
+  if (state && state.macros) {
+    for (var macroName in state.macros) {
+      TEX.Definitions.macros[macroName] = state.macros[macroName];
+      TEX.Definitions.macros[macroName].isUser = true;
```

Comment:
Should not need to set this, as it is already set in the objects that are pointed to in `state.macros`.